packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,692 @@
+    }
+  }
+}
+
+class _RenderSegmentedControl extends RenderBox {
+  _RenderSegmentedControl({
+    @required this.children,
+    @required this.selectedValue,
+    @required this.pressedValue,
+    @required TextDirection textDirection,
+  }) : _textDirection = textDirection;
+
+  List<Widget> children;
+
+  ValueKey<int> selectedValue;
+  ValueKey<int> pressedValue;
+
+  final List<Key>  _hitTestLog = <Key>[];
+
+  final List<Paint> _fillPaint = <Paint>[];
+
+  List<Paint> get fillPaint => _fillPaint;
+
+  final List<RenderBox> _slots = <RenderBox>[];
+
+  final List<RenderBox> _boxes = <RenderBox>[];
+
+  List<RenderBox> get boxes => _boxes;
+
+  final List<RRect> _rects = <RRect>[];
+
+  void _setBox(int slot, RenderBox value) {
+    if (slot >= _slots.length) {
+      boxes.insert(slot, _updateChild(null, value, slot));
+    } else {
+      boxes.insert(slot, _updateChild(boxes.elementAt(slot), value, slot));
+    }
+  }
+
+  RenderBox _updateChild(RenderBox oldChild, RenderBox newChild, int slot) {
+    if (oldChild != null) {
+      dropChild(oldChild);
+      _slots.remove(oldChild);
+    }
+    if (newChild != null) {
+      _slots.insert(slot, newChild);
+      adoptChild(newChild);
+    }
+    return newChild;
+  }
+
+  Iterable<RenderBox> get _children sync* {
+    for (RenderBox box in boxes) {
+      if (box != null) {
+        yield box;
+      }
+    }
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+
+  set textDirection(TextDirection value) {
+    if (_textDirection == value) {
+      return;
+    }
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    for (RenderBox child in _children) {
+      child.attach(owner);
+    }
+  }
+
+  @override
+  void detach() {
+    super.detach();
+    for (RenderBox child in _children) {
+      child.detach();
+    }
+  }
+
+  @override
+  void redepthChildren() {
+    _children.forEach(redepthChild);
+  }
+
+  @override
+  void visitChildren(RenderObjectVisitor visitor) {
+    _children.forEach(visitor);
+  }
+
+  static BoxParentData _boxParentData(RenderBox box) => box.parentData;
+
+  static Size _boxSize(RenderBox box) => box == null ? Size.zero : box.size;
+
+  static Rect _boxRect(RenderBox box) {
+    return box == null ? Rect.zero : _boxParentData(box).offset & box.size;
+  }
+
+  static double _minWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _minHeight(RenderBox box, double width) {
+    return box == null ? 0.0 : box.getMinIntrinsicHeight(width);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    double total = 0.0;
+    for (RenderBox box in boxes) {
+      total += _minWidth(box, height);
+    }
+    return total;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) =>
+      computeMinIntrinsicWidth(height);
```

Comment:
Either make these three be on one line (each), or change them to have curly braces. We don't use the `=>` operator unless it fits on one line.

(and check the rest of the file).